infra/import_graph.py

```diff
@@ -0,0 +1,94 @@
+    node_type that matches the given node.
+
+
+    The SourceCred type system is still pretty ad-hoc,
+    (see: https://github.com/sourcecred/sourcecred/issues/710), so this system is
+    likely to change in the future.
+    """
+    NODE_PREFIX_TO_TYPE = [
+        {"prefix": ["sourcecred", "github", "REPO"], "type": "github/repo"},
+        {"prefix": ["sourcecred", "github", "USERLIKE", "USER"], "type": "github/user"},
+        {"prefix": ["sourcecred", "github", "USERLIKE", "BOT"], "type": "github/bot"},
+        {"prefix": ["sourcecred", "github", "PULL"], "type": "github/pull"},
+        {"prefix": ["sourcecred", "github", "ISSUE"], "type": "github/issue"},
+        {"prefix": ["sourcecred", "github", "REVIEW"], "type": "github/review"},
+        {"prefix": ["sourcecred", "github", "COMMENT"], "type": "github/comment"},
+        {"prefix": ["sourcecred", "git", "COMMIT"], "type": "git/commit"},
+    ]
+    return _type_prefix_match(NODE_PREFIX_TO_TYPE, address)
+
+
+def edge_type(address):
+    """
+    Return a string that identifies the "type" of a SourceCred edge.
+
+    For any anticipated SourceCred edge (i.e. it was supplied by one of the two
+    standard SourceCred plugins, i.e. sourcecred/git and sourcecred/github),
+    this method returns a string which identifies the most specific declared
+    edge_type that matches the given node.
+
+
+    The SourceCred type system is still pretty ad-hoc,
+    (see: https://github.com/sourcecred/sourcecred/issues/710), so this system is
+    likely to change in the future.
+    """
+    EDGE_PREFIX_TO_TYPE = [
+        {"prefix": ["sourcecred", "github", "HAS_PARENT"], "type": "github/hasParent"},
+        {"prefix": ["sourcecred", "github", "REFERENCES"], "type": "github/references"},
+        {"prefix": ["sourcecred", "github", "MENTIONS_AUTHOR"], "type": "github/mentionsAuthor"},
+        {"prefix": ["sourcecred", "github", "AUTHORS"], "type": "github/authors"},
+        {"prefix": ["sourcecred", "github", "PULL"], "type": "github/pull"},
+        {"prefix": ["sourcecred", "github", "ISSUE"], "type": "github/issue"},
+        {"prefix": ["sourcecred", "github", "REVIEW"], "type": "github/review"},
+        {"prefix": ["sourcecred", "github", "COMMENT"], "type": "github/comment"},
+        {"prefix": ["sourcecred", "github", "MERGED_AS"], "type": "github/mergedAs"},
+        {"prefix": ["sourcecred", "github", "REACTS", "HOORAY"], "type": "github/reactsHooray"},
+        {"prefix": ["sourcecred", "github", "REACTS", "THUMBS_UP"], "type": "github/reactsThumbsUp"},
+        {"prefix": ["sourcecred", "github", "REACTS", "HEART"], "type": "github/reactsHeart"},
+        {"prefix": ["sourcecred", "github", "REACTS", "ROCKET"], "type": "github/reactsRocket"},
+        {"prefix": ["sourcecred", "git", "HAS_PARENT"], "type": "git/hasParent"},
+    ]
+    return _type_prefix_match(EDGE_PREFIX_TO_TYPE, address)
+
+
+def jsonToMultiDiGraph(json):
```

Comment:
Python identifiers are snake-case, not camel-case; this should be
`json_to_multidigraph`.

Reply:
Renamed.